pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -90,7 +90,8 @@ class Agent(Generic[AgentDeps, ResultData]):
     be merged with this value, with the runtime argument taking priority.
     """
 
-    _result_schema: _result.ResultSchema[ResultData] | None = field(repr=False)
+    _default_result_schema: _result.ResultSchema[ResultData] | None = field(repr=False)
+    _current_result_schema: _result.ResultSchema[ResultData] | None = field(repr=False)
```

Comment:
ha, this would break a lot of cases since agents can be run concurrently, they need to be static.

You'll need to pass the `result_schema` around everywhere you need to use it.

I suspect a better approach would be to move all the run logic from `Agent` into a new `AgentRun` (data)class, to avoid argument overload.

Reply:
This seems like a good idea structurally to implement sooner rather than later